src/script/lua_api/l_object.cpp

```diff
@@ -187,7 +187,8 @@ int ObjectRef::l_punch(lua_State *L)
 	u16 dst_origin_hp = puncher->getHP();
 
 	// Do it
-	co->punch(dir, &toolcap, puncher, time_from_last_punch);
+	u16 wear = co->punch(dir, &toolcap, puncher, time_from_last_punch);
+	lua_pushnumber(L, wear);
```

Comment:
Needs documentation.

Reply:
is already documented
https://github.com/minetest/minetest/blob/5fa614d97e13af64be490336392abe2a54fdcbc1/doc/lua_api.txt#L1811-L1813

Reply:
Oh weird, this means it never worked before.

Reply:
Indeed